libvmaf/src/feature/motion.c

```diff
@@ -101,176 +131,198 @@ int motion(int (*read_noref_frame)(float *main_data, float *temp_data, int strid
-        // filter
         // apply filtering (to eliminate effects film grain)
         // stride input to convolution_f32_c is in terms of (sizeof(float) bytes)
         // since stride = ALIGN_CEIL(w * sizeof(float)), stride divides sizeof(float)
         // ===============================================================
-        if (next_frame_read)
-        {
-            convolution_f32_c(FILTER_5, 5, next_ref_buf, next_blur_buf, temp_buf, w, h, stride / sizeof(float), stride / sizeof(float));
+        if (next_frame_read) { 
+            convolution_f32_c(FILTER_5, 5, next_ref_buf, next_blur_buf, temp_buf, w, h, stride / sizeof(float), stride / sizeof(float)); 
         }
-
+        
         /* =========== motion ============== */
-
         // compute
-        if (frm_idx == 0)
-        {
-            score = 0.0;
-            score2 = 0.0;
-        }
-        else
-        {
-            if ((ret = compute_motion(prev_blur_buf, blur_buf, w, h, stride, stride, &score)))
-            {
+        if (frm_idx == 0){
+            score = 0.0; 
+        } else {     
+            if ((ret = compute_motion(prev_blur_buf, blur_buf, w, h, stride, stride, &score, pass))){
                 printf("error: compute_motion (prev) failed.\n");
-                fflush(stdout);
+                fflush(stdout); 
                 goto fail_or_end;
-            }
-
-            if (next_frame_read)
-            {
-                if ((ret = compute_motion(blur_buf, next_blur_buf, w, h, stride, stride, &score2)))
-                {
-                    printf("error: compute_motion (next) failed.\n");
-                    fflush(stdout);
-                    goto fail_or_end;
-                }
-                score2 = MIN(score, score2);
-            }
-            else
-            {
-                score2 = score;
-            }
+            }      
         }
-
-        // print
-        printf("motion: %d %f\n", frm_idx, score);
-        printf("motion2: %d %f\n", frm_idx, score2);
         fflush(stdout);
-
         memcpy(prev_blur_buf, blur_buf, data_sz);
         memcpy(ref_buf, next_ref_buf, data_sz);
         memcpy(blur_buf, next_blur_buf, data_sz);
 
-        if (!next_frame_read)
-        {
-            break;
+        if (!next_frame_read) { 
+            break; 
         }
     }
 
-    ret = 0;
-
+    // The second pass (pass 1) is an N^2 comparison of relative motion between all frames
+    // in the input video file. The outer loop frame is referred to as 'b_frame_buf' and 
+    // the inner loop frame is 'c_frame_buf', you can think of 'b' as the reference frame
+    // for all the iteration of 'c' frames to be compared to. Blur buf's serve the same purpose
+    // but have blurring & convolutioin applied to eliminate noise to give more accurate motion scores
+    // b   c
+    // ------
+    // 0   1   
+    // 0   2
+    //  ...
+    // 2   2
+    // 2   3
+    //  ...
+    // 52  52
```

Comment:
technically you shouldn't compare N to N...i.e. 52 and 51 should be the last pair